.github/workflows/main.yml

```diff
@@ -0,0 +1,127 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - '*_maintenance'
+  pull_request:
+    branches:
+      - '*'
+  release:
+      types: [published]
+
+jobs:
+
+  build:
+
+    strategy:
+
+      # Don't cancel other jobs in the build matrix if one job fails.
+      fail-fast: false
+
+      matrix:
+
+        # Rather than generate all permutations of various settings,
+        # we want to explicitly list each of the variants we want to
+        # test. We can use `name` to declare the names of our variants,
+        # and then use `include` to define their settings.
+
+        name: [
+          linux-gcc11,
+          macos-arm64,
+        ]
+
+        include:
+
+          - name: linux-gcc11
+            os: ubuntu-20.04
+            publish: true
+            containerImage: ghcr.io/gafferhq/build/build:3.0.0
+
+          - name: macos-arm64
+            os: macos-14
+            publish: true
+            containerImage:
+            jobs: 3
```

Comment:
It's nice having this as a variable in the matrix. Maybe we should make it compulsory, so instead of the `|| 2` fallback we should specify `jobs` for Linux too?

The [docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) seem to suggest that the Linux runners have 4 cores now, so maybe it's worth bumping jobs to match to see if it helps?

Reply:
`jobs` does seem a natural fit in the matrix, I've now made it compulsory and specified `jobs: 4` for Linux, which has cut another hour from the total build time. We're now down to about 4.5 hours for a full dependencies build on Linux.

Thanks for pointing out the change to available cores on the runners, I was defaulting to 2 out of habit from our other CI configs and it's a decent win bumping this up.